test/reactive-handler.spec.ts

```diff
@@ -791,4 +791,55 @@ describe('ReactiveHandler', () => {
             });
         });
     });
+    describe('with magic key property', () => {
+        it('should support magicPropertyKey', () => {
+            const o = {};
+            const target = new ReactiveMembrane({
+                magicPropertyKey: 'foo',
+            });
+
+            const wet = target.getProxy(o);
+            expect(Object.getOwnPropertyNames(wet).length).toBe(1);
+            expect('foo' in wet).toBe(true);
+            expect('foo' in o).toBe(false);
+            expect(wet.foo).toBe(undefined);
+        });
+        it('should not shadow magicPropertyKey if the target has it', () => {
+            const o = { foo: 1 };
+            const target = new ReactiveMembrane({
+                magicPropertyKey: 'foo',
+            });
+
+            const wet = target.getProxy(o);
+            expect(Object.getOwnPropertyNames(wet).length).toBe(1);
+            expect('foo' in wet).toBe(true);
+            expect('foo' in o).toBe(true);
+            expect(wet.foo).toBe(1);
+        });
+        it('should support frozen target when using magicPropertyKey', () => {
+            const o = Object.freeze({});
+            const target = new ReactiveMembrane({
+                magicPropertyKey: 'foo',
+            });
+
+            const wet = target.getProxy(o);
+            expect(Object.getOwnPropertyNames(wet).length).toBe(1);
+            expect('foo' in wet).toBe(true);
+            expect('foo' in o).toBe(false);
+            expect(wet.foo).toBe(undefined);
+        });
+        it('should support freezing the proxy when using magicPropertyKey', () => {
+            const o = {};
+            const target = new ReactiveMembrane({
+                magicPropertyKey: 'foo',
+            });
+
+            const wet = target.getProxy(o);
+            Object.freeze(wet);
+            expect(Object.getOwnPropertyNames(wet).length).toBe(1);
+            expect('foo' in wet).toBe(true);
+            expect('foo' in o).toBe(false);
+            expect(wet.foo).toBe(undefined);
+        });
+    });
```

Comment:
Add test cases for 
1. Defining a new property using Object.defineProperty on the wet object
2. Delete the magic property key on the wet object
3. Have magic property defined on dry object, define same property on wet object, delete the property on wet object. Recheck if property exists and the value.

Reply:
On top of this list, I would also add test to check the returned property descriptor of the magic propety on the wet object: `{ value: undefined, enumerable: false, configurable: false, writable: false }`

Reply:
@pmdartus this is the test that you asked for, in fact surface a problem, thanks:
https://github.com/salesforce/observable-membrane/pull/48/commits/b2aec8c5695fcbee8e3b1484762b78ad5c9bb215

@ravijayaramappa I don't understand your test... can you elaborate?

Reply:
Assuming that the magic property is discoverable, i was suggesting to add test cases for the corner cases.
```js
const o = {};
const target = new ReactiveMembrane({
    tagPropertyKey: '$$MagicKey$$',
});

const wet = target.getProxy(o);
Object.defineProperty(wet, '$$MagicKey$$', {
    value: 'bar',
    configurable: true
});

#1
expect('$$MagicKey$$' in wet).toBe(true);
expect(object.keys(wet)).toEqual(['$$MagicKey$$']);

#2
delete wet["$$MagicKey$$"]
expect('$$MagicKey$$' in wet).toBe(false); // what should happen here?
expect(object.keys(wet)).toEqual([]); 
```

Reply:
```js
#3
const dry = {};
const target = new ReactiveMembrane({
    tagPropertyKey: '$$MagicKey$$',
});

const wet = target.getProxy(dry);
Object.defineProperty(dry, '$$MagicKey$$', {
    value: 'bar',
    configurable: true
});

Object.defineProperty(wet, '$$MagicKey$$', {
    value: 'baz',
    configurable: true
});

expect('$$MagicKey$$' in wet).toBe(true);
expect(object.keys(wet)).toEqual(['$$MagicKey$$']);
expect(wet['$$MagicKey$$]).toBe('baz');
expect(dry['$$MagicKey$$]).toBe('baz'); // value set on wet propagated to dry

delete wet["$$MagicKey$$"]
expect('$$MagicKey$$' in wet).toBe(false);
expect(object.keys(wet)).toEqual([]);
expect(dry['$$MagicKey$$]).toBeUndefined();

expect('$$MagicKey$$' in dry).toBe(false);
expect(object.keys(dry)).toEqual([]);
expect(dry['$$MagicKey$$]).toBeUndefined();
```

#4 would be the same test setup with the descriptor marked as `configurable: false`